iceoryx_hoofs/test/moduletests/test_posix_signal_watcher.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_hoofs/testing/watch_dog.hpp"
+#include "test.hpp"
+#include <atomic>
+
+namespace
+{
+using namespace ::testing;
+using namespace iox::posix;
+
+class SignalWatcherTester : public SignalWatcher
+{
+  public:
+    SignalWatcherTester() = default;
+};
+
+class SignalWatcher_test : public Test
+{
+  public:
+    void SetUp() override
+    {
+        sut = static_cast<SignalWatcherTester*>(&SignalWatcher::getInstance());
+        sut->~SignalWatcherTester();
+        new (sut) SignalWatcherTester();
+        watchdog.watchAndActOnFailure([] { std::terminate(); });
+    }
+
+    void TearDown() override
+    {
+    }
```

Comment:
Please also reset the `SignalWatcher` on tear down to leave a clean environment behind